sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+import com.azure.containers.containerregistry.models.ListRepositoriesOptions;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * Initializes a new instance of the asynchronous ContainerRegistry type.
+ */
+@ServiceClient(builder = ContainerRegistryClientBuilder.class, isAsync = true)
+public final class ContainerRegistryAsyncClient {
+    private final ContainerRegistriesImpl serviceClient;
+    private final ClientLogger logger = new ClientLogger(ContainerRegistryAsyncClient.class);
+    private static final String CONTINUATIONLINKHEADERNAME = "Link";
+    private static final String LINKMATCHER = "(<(.+)>;.*)";
+    private static final String ORDERBYQUERYPARAM = "&orderby=";
+
+    /**
+     * Initializes an instance of ContainerRegistries client.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    ContainerRegistryAsyncClient(ContainerRegistriesImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * List repositories.
+     *
+     * @return list of repositories.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories() {
+        ListRepositoriesOptions options = new ListRepositoriesOptions();
+        return this.listRepositories(options);
+    }
+
+    /**
+     * List repositories.
+     *
+     * @param options for the list repositories call.
+     * @return list of repositories.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws NullPointerException thrown if the parameter is null.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories(ListRepositoriesOptions options) {
+        if (options == null) {
+            throw logger.logExceptionAsError(new NullPointerException("options can't be null"));
+        }
```

Comment:
Do we want to throw here or simply use a default instance of `ListRepositoriesOptions`?

Reply:
I was looking at our SDK guidelines and it mentioned we should throw exceptions when arguments are null.

Is there a common practice we follow?

---
In reply to: [602629803](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602629803) [](ancestors = 602629803)

Reply:
We throw when null and expecting non-null. From what I could tell here the options class is a property bag that is optional